CMakeLists.txt

```diff
@@ -5,7 +5,7 @@
 ###############################################################################
 # CMake definition.
 
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
Forgot to mention but we should also update the minimum version in other places, half a dozen or so references came up when looking for 3.13 in the code base.

Reply:
Another TSC query or two?

1. Should we change the version mandated by us for the consumer of the project? akin to PRIVATE, vs PUBLIC
2. For files included from external sources should we update those?
3. For CMake files used to build our dependencies
4. For "tests" and other configuration setup where we don't need the feature should we match version requirement